test/integrationtests/username_password_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package integrationtests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/msal"
+)
+
+var labClientInstance = &labClient{}
+
+func AcquireTokenByUsernamePasswordCommonAAD(t *testing.T, user user, authority string, scope []string, appID string) {
+	options := msal.DefaultPublicClientApplicationOptions()
+	options.Authority = authority
+	publicClientApp, err := msal.NewPublicClientApplication(appID, &options)
+	if err != nil {
+		panic(err)
+	}
+	result, err := publicClientApp.AcquireTokenByUsernamePassword(
+		context.Background(),
+		scope,
+		user.Upn,
+		user.Password,
+	)
+	if err != nil {
+		t.Error(err)
+	}
+	if result.AccessToken == "" {
+		t.Error("No access token found")
+	}
+	if result.Account.GetUsername() != user.Upn {
+		t.Errorf("Incorrect user account")
+	}
```

Comment:
Do we want to add checks for others objects in authentication result, like id token?

Reply:
Thats a good suggestion ! I see though that the idToken field is not exposed in the authentication result, will create an issue there as well to track this, once that is resolved, ill add a check here as well.

Reply:
Issue to track this : https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/107